pkg/ingester-rf1/metastore/client/client.go

```diff
@@ -35,9 +41,9 @@ type Client struct {
 	config  Config
 }
 
-func New(config Config) (c *Client, err error) {
+func New(config Config, r prometheus.Registerer) (c *Client, err error) {
```

Comment:
Perhaps just pass in the metrics as a struct instead? That's what we do in the WAL.